nni/compression/pytorch/compressor.py

```diff
@@ -608,24 +626,35 @@ def quant_backward(tensor, grad_output, quant_type):
         tensor
             gradient of the input of quantization operation
         """
-        return grad_output
+        tensor_q = QuantGrad._quantize(tensor, scale, zero_point)
```

Comment:
No need to calculate` tensor_q here`, return a mask from `wrapper.quantizer.quantize_input` for example.

Reply:
No, we need unclampped quantized data, wrapper.quantizer.quantize_input is clampped to [qmin, qmax], so the mask will always be ones everywhere

Reply:
I mean you can do this inside the `wrapper.quantizer.quantize_input` and return the mask which can be used in backward.

Reply:
Therefore you don't do quantization twice and tensor saved will be smaller by 4 times.

Reply:
Yeah, u are right. I did write the code like that.. For doing this ,we need to modify _quantize() to return mask. But i give it up because i dont think it's a good design. Just let _quantize() to return quantized_weight is a better design i think.

Reply:
True, but keeping this design will actually make many quantization algorithms with special backward function harder and less efficient to implement. The actual problem is quant backward is inside the autograd while the forward is inside the quantizer module. I think it will be much better if forward and backward logic is moved into autograd function together and quantizer only maintain some states.

Reply:
What I said is actually what Pytorch implementation does.

Reply:
Yes u r right. And this calls for a HUGE design change with current implementation.

Reply:
And since we've talked about this. I think the current design indeed need to be changed.
1. I agree a lot on what u said that move forward and backward into QuantGrad, and Quantizer only matain the status

Reply:
for now, i think i will just push the correct STE code and what u mentioned is left for next pr~